pkg/hubble/relay/observer.go

```diff
@@ -88,14 +95,47 @@ func (s *Server) GetFlows(req *observerpb.GetFlowsRequest, stream observerpb.Obs
 			}
 		})
 	}
+
+	pq := queue.NewPriorityQueue(qlen)
+	sortedFlows := make(chan *observerpb.GetFlowsResponse, qlen)
+	go func() {
+		defer close(sortedFlows)
+	flowsLoop:
+		for {
+			select {
+			case flow, ok := <-flows:
```

Comment:
should there be a `case <-ctx.Done()` to stop if context is done?

Reply:
I think so too. I understand that `flows` will be closed once `ctx` is cancelled, but there will be some delay before this goroutine will notice it and shut down.

If there are still in `flows`, they will be added to `pq` even if `ctx` is cancelled, as `ctx.Done` is only checked once `pq` is full (or after the timeout).

Reply:
Right, nice catch.